modules/programs/vscode.nix

```diff
@@ -172,5 +182,23 @@ in {
         "${extensionPath}".source = extensionsFolder;
       }))
     ];
+
+    home.activation =
+      mkIf (cfg.mutableUserSettings && cfg.userSettings != { }) {
+        injectVscodeSettings = lib.hm.dag.entryAfter [ "writeBoundary" ] ''
+          tmp="$(mktemp)"
+          if [[ -v DRY_RUN ]]; then
+            echo tmp="\$(mktemp)"
+            echo ${pkgs.jq}/bin/jq -s "'reduce .[] as \$x ({}; . * \$x)'" "${
+              jsonFormat.generate "vscode-user-settings" cfg.userSettings
+            }" "${configFilePath}" ">" "$tmp"
+          else
+            ${pkgs.jq}/bin/jq -s 'reduce .[] as $x ({}; . * $x)' "${
+              jsonFormat.generate "vscode-user-settings" cfg.userSettings
+            }" "${configFilePath}" > "$tmp"
+          fi
+          $DRY_RUN_CMD mv "$tmp" "${configFilePath}"
+        '';
+      };
```

Comment:
This seems overly complicated. I think that it's fine if the dry-run creates temporary files. Just make sure that it doesn't actually override the `configFilePath`.

Regardless, this code needs a comment or two explaining what it's doing.

---

I guess this situation is also a vote in favor of having https://github.com/nix-community/home-manager/pull/1145 merged...